moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -402,68 +402,56 @@ void planning_scene_monitor::CurrentStateMonitor::tfCallback()
 
-  ros::Time latest_updates[multi_dof_joints.size()];
-
-  for (size_t i = 0; i < multi_dof_joints.size(); i++)
-  {
-    const moveit::core::JointModel* joint = multi_dof_joints[i];
-    const std::string& child_frame = joint->getChildLinkModel()->getName();
-    const std::string& parent_frame =
-        joint->getParentLinkModel() ? joint->getParentLinkModel()->getName() : robot_model_->getModelFrame();
-
-    std::string err;
-    if (tf_->getLatestCommonTime(parent_frame, child_frame, latest_updates[i], &err) != tf::NO_ERROR)
-    {
-      ROS_DEBUG_STREAM_THROTTLE(1, "Unable to update multi-dof joint '"
-                                       << joint->getName() << "': tf has no common time between '"
-                                       << parent_frame.c_str() << "' and '" << child_frame.c_str() << "': " << err);
-      latest_updates[i] = ros::Time(0);
-      continue;
-    }
-  }
-
   bool update = false;
   bool changes = false;
-
   {
     boost::mutex::scoped_lock _(state_update_lock_);
 
     for (size_t i = 0; i < multi_dof_joints.size(); i++)
     {
       const moveit::core::JointModel* joint = multi_dof_joints[i];
-
-      if (latest_updates[i] <= joint_time_[joint])
-        continue;
-
       const std::string& child_frame = joint->getChildLinkModel()->getName();
       const std::string& parent_frame =
           joint->getParentLinkModel() ? joint->getParentLinkModel()->getName() : robot_model_->getModelFrame();
 
+      ros::Time latest_common_time;
+      std::string err;
+      if (tf_->getLatestCommonTime(parent_frame, child_frame, latest_common_time, &err) != tf::NO_ERROR)
+      {
+        ROS_WARN_STREAM_THROTTLE(1, "Unable to update multi-DOF joint '"
+                                        << joint->getName() << "': TF has no common time between '"
+                                        << parent_frame.c_str() << "' and '" << child_frame.c_str() << "': " << err);
+        continue;
+      }
+
+      if (latest_common_time <= joint_time_[joint] && latest_common_time > ros::Time(0))
+        continue;
+
       tf::StampedTransform transf;
       try
       {
-        tf_->lookupTransform(parent_frame, child_frame, latest_updates[i], transf);
+        tf_->lookupTransform(parent_frame, child_frame, latest_common_time, transf);
       }
       catch (tf::TransformException& ex)
       {
         ROS_ERROR_STREAM_THROTTLE(1, "Unable to update multi-dof joint '" << joint->getName()
                                                                           << "'. TF exception: " << ex.what());
         continue;
       }
-      joint_time_[joint] = latest_updates[i];
+      joint_time_[joint] = latest_common_time;
 
       Eigen::Affine3d eigen_transf;
       tf::transformTFToEigen(transf, eigen_transf);
 
-      double new_values[joint->getStateSpaceDimension()];
-      joint->computeVariablePositions(eigen_transf, new_values);
+      std::vector<double> new_values(joint->getStateSpaceDimension());
```

Comment:
Why did you change `double[]` to `vector<double>`? The array is created on stack, while the vector requires heap allocation which is slower.

Reply:
When using [cpplint](https://github.com/cpplint/cpplint) I get this error
> Do not use variable-length arrays.  Use an appropriately named ('k' followed by CamelCase) compile-time constant for the size.  [runtime/arrays] [1]

Your comment about speed is valid though, so I will change it back.